src/utils/metadata.py

```diff
@@ -125,68 +129,30 @@ def build_metadata_from_setuptools_dict(metadata):
 
 def get_metadata_from_distribution(distribution_name):
     """Get the metadata from a distribution."""
-    # useful information: https://packaging.python.org/specifications/core-metadata/
-    dist = get_distribution(distribution_name)
-
     metadata = {}
-
-    # get metadata name or error
-    for mdn in METADATA_NAMES:
-        if dist.has_metadata(mdn):
-            metadata_name = mdn
-            break
-    else:
-        raise FileNotFoundError(
-            f"package has no metadata file with name: {METADATA_NAMES}"
-        )
-
-    for mdl in dist.get_metadata_lines(metadata_name):
-        # guard against silly data
-        if ":" not in mdl:
-            continue
-
-        key, value = mdl.split(":", 1)
-        key = key.lower()
-        value = value.strip()
-
+    # useful information: https://packaging.python.org/specifications/core-metadata/
+    # the keys are all lower case and using _ as a separator
+    metadata_json = importlib_metadata.metadata(distribution_name).json
+    for key, value in metadata_json.items():
         # treat UNKNOWN as no value, this the setuptools metadata
         # equivalent of None
-        if value == "UNKNOWN":
+        if not value or value == "UNKNOWN":
             continue
-
         # handle special cases
-        if key == "project-url":
+        if key == "project_url":
             key = "project_urls"
-            inner_key, inner_value = value.split(",", 1)
-            inner_value = inner_value.strip()
-            # project-urls is a dict
-            value = metadata.get(key, {})
-            value[inner_key] = inner_value
-        elif key in {
-            "platform",
```

Comment:
these special cases aren't needed any more?

Reply:
They should be handled automatically when accessing the `.json` property . See:

- https://github.com/python/importlib_metadata/blob/a6038c3f2bddad2a6113deb5358e911071ec71d1/importlib_metadata/_adapters.py#L21
- https://github.com/python/importlib_metadata/blob/a6038c3f2bddad2a6113deb5358e911071ec71d1/importlib_metadata/_adapters.py#L77

We are still adding a trailing `s` to the key for backward compatibility, although I don't see where it's needed.